jetty-util/src/main/java/org/eclipse/jetty/util/ssl/SslContextFactory.java

```diff
@@ -1264,9 +1264,13 @@ protected KeyStore loadTrustStore(Resource resource) throws Exception
         return managers;
     }
 
+    /**
+     * @deprecated use {@link SslContextFactory.Server#newSniX509ExtendedKeyManager(X509ExtendedKeyManager)} instead
+     */
+    @Deprecated
     protected X509ExtendedKeyManager newSniX509ExtendedKeyManager(X509ExtendedKeyManager keyManager)
     {
-        return new SniX509ExtendedKeyManager(keyManager);
+        throw new UnsupportedOperationException("X509ExtendedKeyManager only supported on Server");
```

Comment:
This change seems to have broken creation of a HTTPS connector for those using `SslContextFactory` directly instead of its `Server` subclass, like https://github.com/dropwizard/dropwizard/blob/release/1.3.x/dropwizard-jetty/src/main/java/io/dropwizard/jetty/HttpsConnectorFactory.java#L599

For a codepath that is merely "deprecated", this seems unexpected.

Reply:
Agreed.

Now how to convey this particular method properly.
It's 100% invalid outside of `SslContextFactory.Server` usage, and returning null is 100% invalid as well.

Reply:
@bentmann looking at the code you linked, https://github.com/dropwizard/dropwizard/blob/release/1.3.x/dropwizard-jetty/src/main/java/io/dropwizard/jetty/HttpsConnectorFactory.java#L599

That would not work for Server side SNI based certificates / keystore as it sits right now.
Jetty version 9.4.16.v20190411 is the one that introduced the split for Server vs Client (which fixed client configs impacting server, and server configs impacting client, among which was SNI behaviors and EndpointIdentificationAlgorithm behaviors)